pulsar-testclient/src/main/java/org/apache/pulsar/testclient/ManagedLedgerWriter.java

```diff
@@ -240,7 +238,8 @@ public void run() {
 
                     // Acquire 1 sec worth of messages to have a slower ramp-up
                     rateLimiter.acquire((int) msgRate);
-                    final long startTime = System.currentTimeMillis();
+                    final long startTime = System.nanoTime();
+                    final long testEndTime = startTime + (long) (arguments.testTime * 1e9);
```

Comment:
I recommend using `TimeUnit.SECONDS.toNanos()` instead of `1e9` to make it more understandable to other developers